tests/database/test_header_db.py

```diff
@@ -325,6 +465,210 @@ def test_gap_tracking(headerdb, genesis_header, written_headers, evolving_gaps):
         assert change == evolving_gaps[idx][0]
 
 
+@given(st.lists(
+    st.tuples(
+        # True to insert a header (ie~ remove a gap), False to remove a header (ie~ add a gap)
+        st.booleans(),
```

Comment:
Maybe switch this to:
```py
class GapActions(enum.Enum):
  ADD_GAP = enum.auto()
  FILL_GAP = enum.auto()
  
...
        st.sampled_from(GapActions),
```

Note that the order of the enum matters for example shrinking!